src/Plugin/views/argument_default/Membership.php

```diff
@@ -0,0 +1,174 @@
+
+namespace Drupal\og\Plugin\views\argument_default;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheableDependencyInterface;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\Context\ContextProviderInterface;
+use Drupal\views\Plugin\views\argument_default\ArgumentDefaultPluginBase;
+use Drupal\og\MembershipManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the group memberships from the current context.
+ *
+ * @ViewsArgumentDefault(
+ *   id = "og_group_membership",
+ *   title = @Translation("Group memberships from current user")
+ * )
+ */
+class Membership extends ArgumentDefaultPluginBase implements CacheableDependencyInterface {
+
+  /**
+   * The OG context provider.
+   *
+   * @var \Drupal\Core\Plugin\Context\ContextProviderInterface
+   */
+  protected $ogContext;
+
+  /**
+   * The OG membership manager.
+   *
+   * @var \Drupal\og\MembershipManagerInterface
+   */
+  protected $ogMembership;
+
+  /**
+   * The user to be evaluated.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $ogUser;
+
+  /**
+   * Constructs a new Membership instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $og_context
+   *   The OG context provider.
+   * @param \Drupal\og\MembershipManagerInterface $og_membership
+   *   The OG membership manager.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The user to be evaluated.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ContextProviderInterface $og_context, MembershipManagerInterface $og_membership, AccountInterface $user) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->ogContext = $og_context;
+    $this->ogMembership = $og_membership;
+    $this->ogUser = $user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.context'),
+      $container->get('og.membership_manager'),
+      $container->get('current_user')->getAccount()
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['og_group_membership'] = ['default' => ''];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    $form['og_group_membership'] = [];
```

Comment:
Adding text input for now, we should get the lists of groups & roles.

```suggestion
    $form['og_group_membership'] = [];
    $form['entity_type'] = [
      '#type' => 'text',
      '#title' => $this->t('Entity type'),
      '#default_value' => $this->options['entity_type'],
    ];    
    $form['role_id'] = [
      '#type' => 'text',
      '#title' => $this->t('Role'),
      '#default_value' => $this->options['role_id'],
    ];
```